### PR TITLE
fix goroutine leak in disruption test

### PR DESCRIPTION
**Pull request by Abirdcfly**
Signed-off-by: Abirdcfly <fp544037857@gmail.com>

#### What type of PR is this?
/kind cleanup
/sig testing

#### What this PR does / why we need it:
find some goroutine leak
#### Which issue(s) this PR fixes:

Part of #108483 
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/controller/disruption/disruption.go [hunk 1]
```diff
@@ -359,6 +359,8 @@ func verifyGroupKind(controllerRef *metav1.OwnerReference, expectedKind string,
 func (dc *DisruptionController) Run(ctx context.Context) {
 	defer utilruntime.HandleCrash()
 	defer dc.queue.ShutDown()
+	defer dc.recheckQueue.ShutDown()
+	defer dc.broadcaster.Shutdown()
```
Comment (aojea) [inline pkg/controller/disruption/disruption.go:363]: The problem is that this things panic right now :/

https://github.com/kubernetes/kubernetes/pull/108080

**Comment by Abirdcfly** [comment]
/hold 
wait https://github.com/kubernetes/kubernetes/pull/108080 to be merged.

File: pkg/controller/disruption/disruption.go (hunk 1 above)
Comment (wojtek-t) [reply pkg/controller/disruption/disruption.go:363]: Should be fixed now.

**Comment by Abirdcfly** [comment]
/unhold

https://github.com/kubernetes/kubernetes/pull/108080 has been merged.

File: pkg/controller/disruption/disruption_test.go [hunk 2]
```diff
@@ -146,7 +145,7 @@ var customGVK = schema.GroupVersionKind{
 	Kind:    "customresource",
 }
 
-func newFakeDisruptionController() (*disruptionController, *pdbStates) {
+func newFakeDisruptionController(ctx context.Context) (*disruptionController, *pdbStates) {
```
Comment (wojtek-t) [inline pkg/controller/disruption/disruption_test.go:148]: I'm sorry for nit-picking, but can we maybe switch that to be more unified with what we're doing in other tests?

- there is a function that just creates the controller, e.g.:
https://github.com/kubernetes/kubernetes/blob/master/test/integration/replicaset/replicaset_test.go#L120
- there is a function that actually starts that, e.g. https://github.com/kubernetes/kubernetes/blob/master/test/integration/replicaset/replicaset_test.go#L155
- that function returns destroyFn which is explicitly deferred in the test, e.g.:
https://github.com/kubernetes/kubernetes/blob/master/test/integration/replicaset/replicaset_test.go#L502

I know that your code is technically correct but we would benefit from more consistency across our codebase...

Comment (wojtek-t) [reply pkg/controller/disruption/disruption_test.go:148]: Actually sorry - I realized that those are unit tests, so maybe leaving it here as is is fine for now to stop the bleeding...

**Pull request closed without merging**